tests/unit/core/i18n/test_utils.js

```diff
@@ -509,4 +510,48 @@ describe(__filename, () => {
       expect(i18n.moment('1988-09-22').format('ll')).toEqual('22 sept. 1988');
     });
   });
+
+  describe('formatFilesize', () => {
+    const _formatFilesize = ({ _filesize, i18n = fakeI18n(), size = 123 }) => {
+      return utils.formatFilesize({ _filesize, i18n, size });
+    };
+
+    it('receives an expected string from filesize', () => {
+      // This is a sanity check, verifying that filesize does in fact return
+      // what we expect.
+      // Without it filesize could break and we'd never know.
+      const size = 123;
+      expect(_formatFilesize({ size })).toEqual(`${size} B`);
+    });
+
+    it('formats the number returned by filesize', () => {
+      const size = 1234;
+      const _filesize = sinon.stub().returns(`${size} B`);
+      expect(_formatFilesize({ _filesize, size })).toEqual('1,234 B');
+    });
+
+    it('returns null for an invalid string from filesize', () => {
+      const _filesize = sinon.stub().returns('123');
+      expect(_formatFilesize({ _filesize })).toEqual(null);
+    });
+
+    it('returns null for an invalid unit of measure from filesize', () => {
+      const _filesize = sinon.stub().returns('123 BOB');
+      expect(_formatFilesize({ _filesize })).toEqual(null);
+    });
+
+    it.each(['B', 'KB', 'MB', 'GB', 'TB'])(
+      'calls i18n.sprintf with the expected substitution for size %s',
+      (sizeName) => {
+        const i18n = fakeI18n();
+        const size = '123';
```

Comment:
shouldn't '123' be a number (not a string)?

Reply:
I don't think so. It's used as a string everywhere in the test, and it's also going to be a string inside `formatFilesize` when passed into `sprintf`.

Reply:
yep that makes sense